### DISCUSSION TITLE
Can you use cocotb in WSL with ModelSim for Windows?

### DISCUSSION THREAD

**Discussion by timothyscherer** [cocotb/cocotb#3305] [Q&A] [answered]
Hello! I am currently trying to run the quickstart project. I have WSL with Ubuntu set up with everything I think I need, and it seems to compile successfully, but when it comes time to run I get the following:

```
# ** Error: (vsim-3193) Load of "/home/user/work/tb/.venv/lib/python3.10/site-packages/cocotb/libs/libcocotbvpi_modelsim.so" failed: Bad DLL format.
# ** Error (suppressible): (vsim-PLI-3002) Failed to load PLI object file "/home/user/work/tb/.venv/lib/python3.10/site-packages/cocotb/libs/libcocotbvpi_modelsim.so".
```

My python is 64-bit, and the virtual environment was created inside WSL.
My modelsim is also 64-bit, but installed on 64-bit Windows.
Is it possible to do this?

My assumption at this point is that the cocotb shared object was compiled for linux, and thus can't be loaded into a windows program, is there a workaround here? I am trying to adopt cocotb in a more reusable manner so that others don't need to reinstall modelsim within WSL (EDIT: I am also trying this approach, but Modelsim is not meant for Ubuntu).

We use Cygwin for everything, and this is the first time python is getting added to the equation, so any suggestions are greatly appreciated! Our projects are Makefile based (formerly using Cygwin) so I don't think conda is a viable option, at least from my understanding.

**Comment by themperek** [comment, marked as answer]
>My python is 64-bit, and the virtual environment was created inside WSL.
>My modelsim is also 64-bit, but installed on 64-bit Windows.

As far as I understand this you need to use the Linux Modelsim version or Python Windows version. Interesting is that one can run Windows binaries from WSL (I was not aware of this).

Msys-2 should work https://github.com/cocotb/cocotb/wiki/Tier-2-Setup-Instructions#windows-installation-using-msys2
